merge-v1/schema.graphql

```diff
@@ -1,11 +1,3 @@
-"Delete this when done of course"
```

Comment:
I need to comment everything here because the PR does not show any of the types you created as a change.

- I would not name it `Hodler`. Because a Subgraph is a public, open API, we can't assume every querier understands the `Hodler` meme. I would just do `User` or `Account`
- what does `collectionName` represent?
- i dont understand `rarityPreMerge` or `rarityNow`. probably better if you describe what you are trying to do in person and then we can figure out what to include
- `isBlue` and `isYellow` should be removed, and you should create the following:
```graphql
enum BackgroundColor {
  RED
  BLUE
  WHITE
  BLACK
}

enum MassColor {
  YELLOW
   WHITE
   BLACK
 }
 ```
 and then have as fields :
 
 ```graphql
 massColor: MassColor!
backgroundColor: BackgroundColor!
```

Or something very similar to this. My above suggestion I am only 90% confident in, because I didn't read fully how the colors are obtained from the metadata.

I think you should replace `purchasePrice` and `purchaseDate` with the following:

```graphql
type Sale {
  "Concatenation of tokenID-saleNumber"
  id: ID!
  "First sale is 0 - second is 1 - etc"
  saleNumber: Int!
  "Token ID of the mass"
  tokenID: String!
  "Price it was sold for"
  price: BigInt!
  "Block timestamp" # Can be converted in front end to human readable"
  timestamp: Int!
  "Seller"
  from: Bytes!
  "Buyer"  
  to: Bytes!
}
```
There are probably some more we could add but this is a good start. 
Also, notice how I added in graphql comments above each field. Please do so for the rest of the fields as well, as it helps communicate what each of the fields are doing!

Now, update the entity fields to have:
```graphql
sales: [Sales!]! @derivedFrom(field: tokenID)
```

Combine InitialTotals, CurrentMasses, OriginalMasses types into this:
```graphql
type Merge {
  "Set to 1"
  id: ID!
  "Original amount of mass minted"
  originalMasses: Int!
 "Current amount of mass existing"
  currentMasses: Int!
  "Total mass in the collection"
  totalMass: Int!
}
```
I am unsure what `type MassGroups @entity {` is supposed to be, lets talk about it

Reply:
you could have in `Sale` the following:

```graphql
type Token @entity {
  id
  decimals
  symbol
  name
}
```
then on `Sale`:
```graphql
token: Token!
```
For ETH as a token you could do `0xeeeeeeee.......`

Reply:
The `sales: [Sale!]!` field i spoke of above belongs on a `type MassNFT @entity { ... }`

Reply:
My responses (we discussed this in person so we don't need to really continue this here, this is just for records):

1. Will change `Hodler` to `User`
2. `collectionName` is just the name of the collection 'merge.' This will go into a 'global' entity as discussed for the collection.
3. `rarityPreMerge` and `rarityNow` was discussed and determined to be a subjective rarity measurement and is to be removed. It is too subjective and also is dynamic because quantity of certain mSizes fluctuate as people "play the merge game" and asborb other NFTs.
4. Colors of different masses will be accounted for using `enum` and also respect the metadata format.
5. Will implement `Sale` entity as you have outlined, and clean up comments within the schema as per typical schema format.
6. Will create a global entity `Merge` that encompasses general details for the overall collection.
7. Will remove `MassGroups` entity since front end developers can query the details of each mSize grouping using the data from `User` and other entities.